File: solutions/Question 258/Solution2.py
# Problem: 258. Add Digits
# Difficulty: Easy
# Topics: Math, Simulation, Number Theory
# URL: https://leetcode.com/problems/add-digits/
# Runtime: 0 ms
# Memory: 17.9 MB
# Solved via CodePath Auto-Committer

class Solution:
    def addDigits(self, num: int) -> int:
        a = num
        while a>9:
            a=0
            while num>0:
                a+=num%10
                num = num//10
            num = a
        return a
        

File: solutions/Question 206/Solution2.cpp
/*
 * Problem: 206. Reverse Linked List
 * Difficulty: Easy
 * Topics: Linked List, Recursion
 * URL: https://leetcode.com/problems/reverse-linked-list/
 * Runtime: 0 ms
 * Memory: 13.3 MB
 * Solved via CodePath Auto-Committer
 */

class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* prev = nullptr;
        ListNode* curr = head;
        while (curr != nullptr) {
            ListNode* temp = curr->next; 
            curr->next = prev;
            prev = curr;
            curr = temp;
        }
        return prev;
    }
};